src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -156,6 +245,113 @@ public abstract partial class JsonNamingPolicy
+        [System.CLSCompliantAttribute(false)]
+        public bool TryGetUInt32(out uint value) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public bool TryGetUInt64(out ulong value) { throw null; }
+    }
+    public partial class JsonObject : System.Text.Json.JsonNode, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>>, System.Collections.IEnumerable
+    {
+        public JsonObject() { }
+        public JsonObject(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>> jsonProperties) { }
+        public System.Text.Json.JsonNode this[string key] { get { throw null; } set { } }
+        public System.Collections.Generic.ICollection<string> PropertyNames { get { throw null; } }
+        public System.Collections.Generic.ICollection<System.Text.Json.JsonNode> Values { get { throw null; } }
+        public void Add(System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode> jsonProperty) { }
+        public void Add(string propertyName, bool propertyValue) { }
+        public void Add(string propertyName, byte propertyValue) { }
+        public void Add(string propertyName, System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode> propertyValue) { }
+        public void Add(string propertyName, double propertyValue) { }
+        public void Add(string propertyName, short propertyValue) { }
+        public void Add(string propertyName, int propertyValue) { }
+        public void Add(string propertyName, long propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, sbyte propertyValue) { }
+        public void Add(string propertyName, float propertyValue) { }
+        public void Add(string propertyName, string propertyValue) { }
+        public void Add(string propertyName, System.Text.Json.JsonArray propertyValue) { }
+        public void Add(string propertyName, System.Text.Json.JsonNode propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, ushort propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, uint propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, ulong propertyValue) { }
+        public void AddRange(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>> jsonProperties) { }
+        public bool ContainsProperty(string propertyName) { throw null; }
+        public System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode> GetAllPrimaryTypedValues() { throw null; }
+        public System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode> GetAllValuesByPropertyName(string propertyName) { throw null; }
```

Comment:
Should this have a bool to say if it would be recursive?

Reply:
I assumed it to be always recursive as `JsonObject` itself cannot have multiple properties with the same name. So for non-recursive call `GetProperty` can be used.

Reply:
Oh, I should have made `GetProperty` return `JsonNode`, not `JsonObject`, and then probably add a method `GetJsonObject` returning `JsonObject` to avoid casting when accessing nested objects?

Reply:
> I assumed it to be always recursive ...

Recursion should be used lightly.  What's the scenario for getting everything with the same property name at varying depths?  I can understand at a specific depth, but that's really just making sure that we work well with the System.Linq filters:

```C#
foreach (JsonObject obj in someObj.Values.OfType<JsonObject>().Where(child => child.TryGetProperty("color", out JsonNode prop) && prop.Type == JsonValueType.String && ((JsonString)prop).Value == "blue"))
{
     ...
}
```

Reply:
I was thinking of `TestAquiringAllPropertiesValuesWithSpecificName`, but here again, I wondered if we need so specialized methods.